### PR TITLE
feat(contracts): export handler authoring helpers

### PR DESCRIPTION
**Pull request by bonomat**
exporting them lets authors of custom ContractHandlers implement selectPath/getSpendablePaths consistently instead of reimplementing BIP-68/BIP-65 timelock-maturity logic.

## Summary by CodeRabbit

* **New Features**
  * Two contract utility helpers are now exposed in the TypeScript SDK public API.
  * Updated release note: corrected previous count — two helpers were added and are publicly available.

### CODE REVIEW

**Review by ghost — APPROVED** [review]
## Review: feat(contracts): export handler authoring helpers

**Verdict: APPROVE** ✅

### What this PR does
Exports 3 existing internal helpers (`isCsvSpendable`, `isCltvSatisfied`, `resolveRole`) from the barrel `index.ts` so custom `ContractHandler` authors can reuse canonical BIP-65/BIP-68 timelock logic instead of reimplementing it.

### Changes reviewed
- `packages/ts-sdk/src/index.ts` — 5 lines: 1 import + 3 re-exports + 1 comment. **Purely additive. No behavioral changes.**

### Full file review: `contracts/handlers/helpers.ts`

Read the complete implementation (119 lines). All three functions are sound:

- **`isCsvSpendable`** (L99-118): Correctly converts BIP-68 sequence via `sequenceToTimelock`, handles both block-based and time-based relative timelocks, fails safe (returns `false`) on missing context fields. ✅
- **`isCltvSatisfied`** (L87-94): Correct BIP-65 threshold at `500_000_000n`, proper block-height vs timestamp dispatch, fails safe on missing `blockHeight`. ✅
- **`resolveRole`** (L24-70): Correct precedence (explicit role → descriptor → deprecated pubkey fallback). Properly short-circuits when descriptor resolves but doesn't match (L101-113 in test confirms this). HD descriptor edge case handled via `extractRawPubKey` returning `undefined`. ✅

### Cross-repo impact
Grepped all repos under `/root/arkana/repos/` for `isCsvSpendable`, `isCltvSatisfied`, `resolveRole` — **zero downstream consumers today**. This export is net-new public surface with no breakage risk.

### Test coverage
`test/contracts/handlers.test.ts` covers:
- `resolveRole`: 4 test cases (descriptor match, HD fallback to pubkey, unresolvable, no-fallback-when-resolved)
- `isCsvSpendable`: tested indirectly via Default/Delegate/VHTLC handler tests (mature vs not-mature assertions)
- `isCltvSatisfied`: tested indirectly via VHTLC CLTV gating tests (block-height and timestamp variants, missing blockHeight)

Coverage is adequate for the export-only change. No new code paths introduced.

### Protocol safety
These are **read-only predicate functions** — they check timelock conditions but do not sign, mutate VTXOs, or interact with the round lifecycle. Not protocol-critical.

### One note for maintainers
These are now public API. Changing their signatures will be a semver-breaking change. The current signatures are clean and minimal — no concerns there.

LGTM. Ship it.

🤖 Reviewed by Arkana

**Review by ghost — APPROVED** [review]
## Review: feat(contracts): export handler authoring helpers

**Verdict: APPROVE** ✅

### What this PR does
Exports 3 existing internal helpers (`isCsvSpendable`, `isCltvSatisfied`, `resolveRole`) from the barrel `index.ts` so custom `ContractHandler` authors can reuse canonical BIP-65/BIP-68 timelock logic instead of reimplementing it.

### Changes reviewed
- `packages/ts-sdk/src/index.ts` — 5 lines: 1 import + 3 re-exports + 1 comment. **Purely additive. No behavioral changes.**

### Full file review: `contracts/handlers/helpers.ts`

Read the complete implementation (119 lines). All three functions are sound:

- **`isCsvSpendable`** (L99-118): Correctly converts BIP-68 sequence via `sequenceToTimelock`, handles both block-based and time-based relative timelocks, fails safe (returns `false`) on missing context fields. ✅
- **`isCltvSatisfied`** (L87-94): Correct BIP-65 threshold at `500_000_000n`, proper block-height vs timestamp dispatch, fails safe on missing `blockHeight`. ✅
- **`resolveRole`** (L24-70): Correct precedence (explicit role → descriptor → deprecated pubkey fallback). Properly short-circuits when descriptor resolves but doesn't match (L101-113 in test confirms this). HD descriptor edge case handled via `extractRawPubKey` returning `undefined`. ✅

### Cross-repo impact
Grepped all repos under `/root/arkana/repos/` for `isCsvSpendable`, `isCltvSatisfied`, `resolveRole` — **zero downstream consumers today**. This export is net-new public surface with no breakage risk.

### Test coverage
`test/contracts/handlers.test.ts` covers:
- `resolveRole`: 4 test cases (descriptor match, HD fallback to pubkey, unresolvable, no-fallback-when-resolved)
- `isCsvSpendable`: tested indirectly via Default/Delegate/VHTLC handler tests (mature vs not-mature assertions)
- `isCltvSatisfied`: tested indirectly via VHTLC CLTV gating tests (block-height and timestamp variants, missing blockHeight)

Coverage is adequate for the export-only change. No new code paths introduced.

### Protocol safety
These are **read-only predicate functions** — they check timelock conditions but do not sign, mutate VTXOs, or interact with the round lifecycle. Not protocol-critical.

### One note for maintainers
These are now public API. Changing their signatures will be a semver-breaking change. The current signatures are clean and minimal — no concerns there.

LGTM. Ship it.

🤖 Reviewed by Arkana

File: packages/ts-sdk/src/index.ts [hunk 1]
```diff
@@ -448,6 +449,10 @@ export {
     contractFromArkContractWithAddress,
     isArkContract,
     isDiscoverable,
+    // Contract handler authoring helpers (spending-path selection)
+    isCsvSpendable,
+    isCltvSatisfied,
+    resolveRole,
```
Comment (Kukks) [inline packages/ts-sdk/src/index.ts]: please dont export resereRole. This is very specific to the htlc contract

**Review by Kukks — CHANGES_REQUESTED** [review]

**Review by ghost — APPROVED** [review]
**Iterative review — delta from ae23245 → 06b4197**

Only change: `resolveRole` was dropped from the import/export. The PR now exports only the two timelock helpers (`isCsvSpendable`, `isCltvSatisfied`), which is a stricter scope than what I previously approved.

- ✅ Narrower public API surface — good call keeping `resolveRole` internal for now
- ✅ No new code paths, no behavioral changes
- ✅ Previous review findings (helpers.ts correctness, cross-repo impact, test coverage) still apply

Ship it.

🤖 Reviewed by Arkana

**Review by pietro909 — APPROVED** [review]

**Comment by bonomat** [comment]
rebased and force-pushed.
